execution_context.go

```diff
@@ -0,0 +1,174 @@
+	txCheckInterval time.Duration,
+	txType uint8,
+	from, to common.Address,
+	value *big.Int,
+	gasLimit uint64, extraGasLimit uint64,
+	gasPrice float64, extraGasPrice float64,
+	tipCapGwei float64, extraTipCapGwei float64,
+	maxGasPrice float64, maxTipCap float64,
+	data []byte,
+	network networks.Network,
+	beforeSignAndBroadcastHook Hook,
+	afterSignAndBroadcastHook Hook,
+	abis []abi.ABI,
+	gasEstimationFailedHook GasEstimationFailedHook,
+) *TxExecutionContext {
+	// Validate inputs
+	if numRetries < 0 {
+		numRetries = 0
+	}
+	if sleepDuration <= 0 {
+		sleepDuration = DefaultSleepDuration
+	}
+	if txCheckInterval <= 0 {
+		txCheckInterval = DefaultTxCheckInterval
+	}
+
+	// Validate addresses
+	if from == (common.Address{}) {
+		panic("from address cannot be zero")
+	}
+
+	// Validate network
+	if network == nil {
+		panic("network cannot be nil")
+	}
+
+	// Initialize value if nil
+	if value == nil {
+		value = big.NewInt(0)
+	}
+
+	return &TxExecutionContext{
+		actualRetryCount:           0,
+		numRetries:                 numRetries,
+		sleepDuration:              sleepDuration,
+		txCheckInterval:            txCheckInterval,
+		txType:                     txType,
+		from:                       from,
+		to:                         to,
+		value:                      value,
+		gasLimit:                   gasLimit,
+		extraGasLimit:              extraGasLimit,
+		retryGasPrice:              gasPrice,
+		extraGasPrice:              extraGasPrice,
+		retryTipCap:                tipCapGwei,
+		extraTipCapGwei:            extraTipCapGwei,
+		maxGasPrice:                maxGasPrice,
+		maxTipCap:                  maxTipCap,
+		data:                       data,
+		network:                    network,
+		oldTxs:                     make(map[string]*types.Transaction),
+		retryNonce:                 nil,
+		beforeSignAndBroadcastHook: beforeSignAndBroadcastHook,
+		afterSignAndBroadcastHook:  afterSignAndBroadcastHook,
+		abis:                       abis,
+		gasEstimationFailedHook:    gasEstimationFailedHook,
+	}
+}
+
+// adjustGasPricesForSlowTx adjusts gas prices when a transaction is slow
+// Returns true if adjustment was applied, false if limits were reached
+func (ctx *TxExecutionContext) adjustGasPricesForSlowTx(tx *types.Transaction) bool {
+	if tx == nil {
+		return false
+	}
+
+	// Increase gas price by configured percentage
+	currentGasPrice := jarviscommon.BigToFloat(tx.GasPrice(), 9)
+	newGasPrice := currentGasPrice * GasPriceIncreasePercent
+	
+	// Check if new gas price would exceed the caller-defined maximum
+	if ctx.maxGasPrice > 0 && newGasPrice > ctx.maxGasPrice {
+		// Gas price would exceed limit - stop trying
+		return false
+	}
+	
+	// Calculate the retry gas price by subtracting the extra
+	if newGasPrice > ctx.extraGasPrice {
+		ctx.retryGasPrice = newGasPrice - ctx.extraGasPrice
+	} else {
+		// If the new gas price is less than extra, use the current gas price
+		ctx.retryGasPrice = currentGasPrice
+	}
```

Comment:
The idea of extraGasPrice is it allows users to configure an additional gas price to add ontop if an automated mechanism to query gas price from some where else (like a node, a gas price service...).

In this case, since the new gas price is already calculated by f(tx.GasPrice), it includes extraGasPrice already, i think we can just ignore completely `extraGasPrice` here.